react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -3,27 +3,25 @@ import { useState } from 'react';
 export default function ValidatedInput() {
   const [pw, setPw] = useState('');
   const [isValid, setIsValid] = useState(false);
-  const [attempted, setAttempted] = useState(false);
   const [err, setErr] = useState([]);
 
   function validate(e) {
     const allReqs = /(?=.*[!@#$%^&*()])(?=.*\d)(?=.*[A-Z]).*/;
     const typedPw = e.target.value;
     const check = allReqs.test(typedPw) && (typedPw.length > 7);
-
+    let message = [];
     setIsValid(check);
     setPw(typedPw);
-    setAttempted(true);
 
     if (!check) {
-      const msgStyle = { margin: "10px", color: "red" };
-      if (attempted && typedPw.length === 0) {
-        return setErr(<p style={msgStyle}>Password is a required field.</p>);
+      // --- Password Required ---
+      const reqPw = 'Password is a required field.';
+      if (typedPw.length === 0) {
+        message.push(reqPw);
+        return setErr(message);
```

Comment:
I find the code more clear if you
```
setErr(message);
return;
```
since `setErr` doesn't return anything and `validate` doesn't need to return anything.
(TypeScript will complain about the code you have written.)